components/category/Category.js

```diff
@@ -23,16 +23,13 @@ export default function Category({ category, listId, hasDisabledItems }) {
     )
       return;
 
-    updateListItemIndex(destination, source);
+    updateListItemIndex(destination, source, hasDisabledItems);
   };
 
-  //activeListItems
-  const activeListItems = listItems.filter(
-    (listItem) => listItem.categoryId === category.id && !listItem.disabled
-  );
-  //inactiveListItems
-  const inactiveListItems = listItems.filter(
-    (listItem) => listItem.categoryId === category.id && listItem.disabled
+  const filteredListItems = listItems.filter(
+    (listItem) =>
+      listItem.categoryId === category.id &&
+      listItem.disabled === hasDisabledItems
```

Comment:
if disabled has only one of two strings, you may want to set it to true/false instead